### PR TITLE
refactor & add CI

### PR DESCRIPTION
**Pull request by Borda**
make the code as a package and add a CI testing

### CODE REVIEW

File: inspect_data.ipynb [hunk 1]
```diff
@@ -39,11 +39,11 @@
...
     "from matplotlib.patches import Polygon\n",
     "\n",
-    "import utils\n",
-    "import visualize\n",
-    "from visualize import display_images\n",
-    "import model as modellib\n",
-    "from model import log\n",
+    "from mask_rcnn import utils\n",
+    "from mask_rcnn import visualize\n",
+    "from mask_rcnn from visualize import display_images\n",
```
Comment (jmtatsch) [inline inspect_data.ipynb]: from mask_rcnn from visualize construct seems odd?

File: inspect_model.ipynb [hunk 2]
```diff
@@ -35,11 +35,11 @@
...
     "import matplotlib.patches as patches\n",
     "\n",
-    "import utils\n",
-    "import visualize\n",
-    "from visualize import display_images\n",
-    "import model as modellib\n",
-    "from model import log\n",
+    "from mask_rcnn import utils\n",
+    "from mask_rcnn import visualize\n",
+    "from mask_rcnn from visualize import display_images\n",
```
Comment (jmtatsch) [inline inspect_model.ipynb]: from mask_rcnn from visualize construct seems odd?

File: inspect_data.ipynb (hunk 1 above)
Comment (Borda) [reply inspect_data.ipynb]: yes, but it was there before... :) OK, I will correct it

File: inspect_model.ipynb (hunk 2 above)
Comment (Borda) [reply inspect_model.ipynb]: yes, but it was there before... :) OK, I will correct it

**Comment by Borda** [comment]
@jmtatsch , is it fine now?

**Comment by jmtatsch** [comment]
@Borda looks good to me but I am merely another contributor :)
Adding continuous integration would help keep/improve the already good quality code & documentation, guarantee a well working/supported Mask_RCNN implementation that is really useful for the whole Keras community and be a excellent showcase for @waleedka's/matterports skills & professionalism.

**Comment by Borda** [comment]
@waleedka  @philferriere   what do you think?

**Comment by waleedka** [comment]
@Borda This is great. Thank you! 

I haven't gotten the chance to review it in depth yet. I'll do it this week. But I do wonder what the Travis CI is verifying in this case. I'm a fan of CI, but I also want to keep the code as simple and accessible as possible, and CI is adding several new files. So it'll be a balance between the added value and the added complexity. Do you think a subset of this PR without CI would still be useful?

**Comment by Borda** [comment]
Well you can still add code without testing, that is no problem, the CI just check that the sections of code you have covered by test (unittest or doctest which I like rather) won't be "damaged" development or at least it simplifies to fund the breaking points... also it should simplify PR process, because the whole pack of code is checked before any further steps... And last but not least if you, as a user, follow the code sequence in CI config you have kind of guarantee that you can run the package on your computer :)

**Comment by Borda** [comment]
@waleedka so shall it be merged?

**Comment by Borda** [comment]
@waleedka so shall it be merged?

**Comment by Borda** [comment]
@waleedka just as you are the owner of this repo, you should update the key in Travis for Codacy and CodeCov... to get the keys should be a manner of a few clicks it is straightforward :)

**Comment by Borda** [comment]
@waleedka I  don't feel like keeping this PR always updated to master and ready for merge until I hear from you "let's go for a CI and testing"... so please let me know

**Comment by waleedka** [comment]
@Borda I'm really sorry for the slow response. I didn't see your last replies until now, they got buried in my inbox. My apology.

I finally have time to catch up on PRs and other issues. I want to merge this today or tomorrow. I'm planning to move a few files around to simplify the project structure, so I'd like to merge this PR before I make the changes. 

May I ask you to move the CI updates to a separate PR? I haven't used Travis, so before I merge that part I'll need to get familiar with Travis and setup an account, which would delay me a bit.

**Comment by Borda** [comment]
Sure, no problem. Only now I am only on mobile so I'll do it on Tuesday. About the CI can I asistent you with the settings...?

**Comment by hadim** [comment]
I agree that CI should be addressed in another PR. Could I also suggest to move the notebook files in a specific folder? (`notebooks` for example).

**Comment by hadim** [comment]
I just noticed the `setup.py` file is missing from the root of the repository. Would it be possible to add it?

**Comment by waleedka** [comment]
@Borda Thanks! I'll wait for your update

@hadim Often notebooks and Python files are part of the same sample. There is a `samples` directory right now, with only one sample project. After @borda's update I'm going to move the coco and train_shapes files under that, in their own separate folders. Coupled with this PR, that should clean up the directory structure significantly.

**Comment by hadim** [comment]
Make sense to me ! (don't forget about the `setup.py` file).

**Comment by Borda** [comment]
Ok, I will split the PR such way
 * CI testing and code quality
 * packaging & setup.py

**Pull request closed without merging**